vmware-event-router/hack/create_k8s_config.sh

```diff
@@ -0,0 +1,193 @@
+#!/bin/bash -eu
+# Copyright 2019 VMware, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-2
+
+black='\E[30;40m'
+red='\E[31;40m'
+green='\E[32;40m'
+magenta='\E[35;40m'
+cyan='\E[36;40m'
+reset='\033[00m'
+
+cecho () {
+    local default_msg="No message passed."
+
+    message=${1:-$default_msg}
+    color=${2:-$black}
+
+    echo -e "$color"
+    echo -e "$message"
+    tput sgr0
+
+    return
+}
+
+cecho "Enter the following values for deployment:" $cyan
+
+echo -e "${magenta}"
+read -p "vCenter Server FQDN: " VCENTER_SERVER
+if [[ -z "${VCENTER_SERVER}" ]]; then
+    cecho "Please start over. No input entered." $red
+    exit 1
+fi
+
+echo -e "${magenta}"
+read -p "vCenter Server Username: " VCENTER_USERNAME
+if [[ -z "${VCENTER_USERNAME}" ]]; then
+    cecho "Please start over. No input entered." $red
+    exit 1
+fi
+
+echo -e "${magenta}"
+echo -n "vCenter Server Password: "
+read -s VCENTER_PASSWORD
+if [[ -z "${VCENTER_PASSWORD}" ]]; then
+    cecho "Please start over. No input entered." $red
+    exit 1
+fi
+echo ""
+
+echo -e "${magenta}"
+echo -n "OpenFaaS Admin Password: "
+read -s OPENFAAS_PASSWORD
+if [[ -z "${OPENFAAS_PASSWORD}" ]]; then
+    cecho "Please start over. No input entered." $red
+    exit 1
+fi
+echo ""
+
+cecho "Please confirm the following settings are correct:\n" $cyan
+echo -e "\tVCENTER_SERVER=${VCENTER_SERVER}"
+echo -e "\tVCENTER_USERNAME=${VCENTER_USERNAME}"
+echo -e "\tVCENTER_PASSWORD=${VCENTER_PASSWORD}"
+echo -e "\tOPENFAAS_PASSWORD=${OPENFAAS_PASSWORD}"
+
+echo -e "${cyan}"
+read -p "Do you want to proceed with the VEBA K8s deployment [y]: " answer
+case $answer in
+    [Yy]* ) cecho "Starting Deployment ..." $green;;
+    [Nn]* ) echo "Exiting ..." $red; exit;;
+    * ) echo "Exiting ..." $red; exit;;
+esac
+
+
+cecho "Creating vmware namespace" $green
+echo -e "\tkubectl create namespace vmware\n"
+kubectl create namespace vmware
```

Comment:
Can we add a safety check to check for the correct credentials? Perhaps not everyone runs this as `ClusterAdmin` so we could check for the correct permissions with [`kubectl auth can-i`](
https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#-em-can-i-em-) to check for the ability to create namespaces, deployments, secrets, etc. At least for namespaces where we then could assume the others would work as well.

Reply:
Done